openshift-kube-apiserver/filters/deprecatedapirequest/apiaccess_count_controller_test.go

```diff
@@ -754,6 +582,157 @@ func TestSetRequestCountsForNode(t *testing.T) {
 
 }
 
+func TestPersistRequestCountForAllResources(t *testing.T) {
+
+	type logRequestFn func(*controller)
+
+	testCases := []struct {
+		name        string
+		currentHour int
+		existing    []runtime.Object
+		requests    []logRequestFn
+		expected    []*apiv1.APIRequestCount
+	}{
+		{
+			name: "Noop",
+		},
+		{
+			name: "EmptyStatus",
+			existing: []runtime.Object{
+				apiRequestCount("test.v1.group"),
+			},
+			expected: []*apiv1.APIRequestCount{
+				apiRequestCount("test.v1.group", withStatus(
+					withRequestLastHour(withPerNodeAPIRequestLog("node10")),
+					withRequestLast24hN("0,2-23", withPerNodeAPIRequestLog("node10")),
+				)),
+			},
+		},
+		{
+			name: "OnRestart",
+			existing: []runtime.Object{
+				// current hour is 0, this api has not been requested since hour 20
+				apiRequestCount("test.v1.group",
+					withStatus(
+						withRequestLastHour(
+							withPerNodeAPIRequestLog("node10",
+								withPerUserAPIRequestCount("user10", "agent10", withRequestCount("get", 100)),
+							),
+						),
+						withRequestLast24hN("*", withPerNodeAPIRequestLog("node10")),
+						withRequestLast24h(20, withPerNodeAPIRequestLog("node10",
+							withPerUserAPIRequestCount("user10", "agent10", withRequestCount("get", 100)),
+						)),
+						setRequestCountTotals,
+					),
+				),
+				// this api will have some current requests
+				apiRequestCount("test.v2.group"),
+			},
+			requests: []logRequestFn{
+				withRequestN("test.v2.group", 0, "user10", "agent10", "get", 53),
+				withRequestN("test.v3.group", 0, "user10", "agent10", "get", 57),
+			},
+			expected: []*apiv1.APIRequestCount{
+				apiRequestCount("test.v1.group",
+					withStatus(
+						withRequestLastHour(withPerNodeAPIRequestLog("node10")),
+						withRequestLast24hN("0,2-23", withPerNodeAPIRequestLog("node10")),
+						withRequestLast24h(20, withPerNodeAPIRequestLog("node10",
```

Comment:
if we ran this test for each hour leading back to 20, the idea is that I should see this cleared, right?

Reply:
Added unit test.